packages/tools/src/tools/annotation/PlanarFreehandROITool.ts

```diff
@@ -0,0 +1,446 @@
+ * support for oblique views is possible, but not yet supported.
+ *
+ * The resulting annotation's data and metadata (the
+ * state of the viewport while drawing was happening) will get added to the
+ * ToolState manager and can be accessed from the ToolState by calling getAnnotations
+ * or similar methods.
+ *
+ * ```js
+ * cornerstoneTools.addTool(PlanarFreehandROITool)
+ *
+ * const toolGroup = ToolGroupManager.createToolGroup('toolGroupId')
+ *
+ * toolGroup.addTool(PlanarFreehandROITool.toolName)
+ *
+ * toolGroup.addViewport('viewportId', 'renderingEngineId')
+ *
+ * toolGroup.setToolActive(PlanarFreehandROITool.toolName, {
+ *   bindings: [
+ *    {
+ *       mouseButton: MouseBindings.Primary, // Left Click
+ *     },
+ *   ],
+ * })
+ * ```
+ *
+ * Read more in the Docs section of the website.
+ */
+class PlanarFreehandROITool extends AnnotationTool {
+  static toolName = 'PlanarFreehandROI';
+
+  public touchDragCallback: any;
+  public mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  private commonData?: PlanarFreehandROICommonData;
+  isDrawing = false;
+  isEditingClosed = false;
+  isEditingOpen = false;
+
+  private activateDraw: (
+    evt: EventTypes.MouseDownActivateEventType,
+    annotation: PlanarFreehandROIAnnotation,
+    viewportIdsToRender: string[]
+  ) => void;
+  private activateClosedContourEdit: (
+    evt: EventTypes.MouseDownActivateEventType,
+    annotation: PlanarFreehandROIAnnotation,
+    viewportIdsToRender: string[]
+  ) => void;
+  private activateOpenContourEdit: (
+    evt: EventTypes.MouseDownActivateEventType,
+    annotation: PlanarFreehandROIAnnotation,
+    viewportIdsToRender: string[]
+  ) => void;
+  private activateOpenContourEndEdit: (
+    evt: EventTypes.MouseDownActivateEventType,
+    annotation: PlanarFreehandROIAnnotation,
+    viewportIdsToRender: string[]
+  ) => void;
+
+  private renderContour: (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: any,
+    annotation: PlanarFreehandROIAnnotation
+  ) => void;
+  private renderContourBeingDrawn: (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: any,
+    annotation: PlanarFreehandROIAnnotation
+  ) => void;
+  private renderClosedContourBeingEdited: (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: any,
+    annotation: PlanarFreehandROIAnnotation
+  ) => void;
+  private renderOpenContourBeingEdited: (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: any,
+    annotation: PlanarFreehandROIAnnotation
+  ) => void;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        shadow: true,
+        preventHandleOutsideImage: false,
+        allowOpenContours: true,
+        // Radius in canvas coordinates used to join contours.
+        closeContourProximity: 10,
+        subPixelResolution: 4,
```

Comment:
can we add comment on subPixelResolution?

Reply:
Done.